tests/test_dduf.py

```diff
@@ -0,0 +1,180 @@
+import json
+import zipfile
+from pathlib import Path
+from typing import Iterable, Tuple, Union
+
+import pytest
+from pytest_mock import MockerFixture
+
+from huggingface_hub.errors import DDUFCorruptedFileError, DDUFExportError, DDUFInvalidEntryNameError
+from huggingface_hub.serialization._dduf import (
+    DDUFEntry,
+    _validate_dduf_entry_name,
+    export_entries_as_dduf,
+    export_folder_as_dduf,
+    read_dduf_file,
+)
+
+
+class TestDDUFEntry:
+    @pytest.fixture
+    def dummy_entry(self, tmp_path: Path) -> DDUFEntry:
+        dummy_dduf = tmp_path / "dummy_dduf.dduf"
+        dummy_dduf.write_bytes(b"somethingCONTENTsomething")
+        return DDUFEntry(filename="dummy.json", length=7, offset=9, dduf_path=dummy_dduf)
+
+    def test_dataclass(self, dummy_entry: DDUFEntry):
+        assert dummy_entry.filename == "dummy.json"
+        assert dummy_entry.length == 7
+        assert dummy_entry.offset == 9
+        assert str(dummy_entry.dduf_path).endswith("dummy_dduf.dduf")
+
+    def test_read_text(self, dummy_entry: DDUFEntry):
+        assert dummy_entry.read_text() == "CONTENT"
+
+    def test_as_mmap(self, dummy_entry: DDUFEntry):
+        with dummy_entry.as_mmap() as mmap:
+            assert mmap == b"CONTENT"
+
+
+class TestUtils:
+    @pytest.mark.parametrize("filename", ["dummy.txt", "dummy.json", "dummy.safetensors"])
+    def test_entry_name_valid_extension(self, filename: str):
+        assert _validate_dduf_entry_name(filename) == filename
+
+    @pytest.mark.parametrize("filename", ["dummy", "dummy.bin", "dummy.dduf", "dummy.gguf"])
+    def test_entry_name_invalid_extension(self, filename: str):
+        with pytest.raises(DDUFInvalidEntryNameError):
+            _validate_dduf_entry_name(filename)
+
+    @pytest.mark.parametrize("filename", ["encoder\\dummy.json", "C:\\dummy.json"])
+    def test_entry_name_no_windows_path(self, filename: str):
+        with pytest.raises(DDUFInvalidEntryNameError):
+            _validate_dduf_entry_name(filename)
+
+    def test_entry_name_stripped(
+        self,
+    ):
+        assert _validate_dduf_entry_name("/dummy.json") == "dummy.json"
+
+    def test_entry_name_no_nested_directory(self):
+        _validate_dduf_entry_name("bar/dummy.json")  # 1 level is ok
+        with pytest.raises(DDUFInvalidEntryNameError):
+            _validate_dduf_entry_name("foo/bar/dummy.json")  # not more
+
+
+class TestExportFolder:
+    @pytest.fixture
+    def dummy_folder(self, tmp_path: Path):
+        folder_path = tmp_path / "dummy_folder"
+        folder_path.mkdir()
+        encoder_path = folder_path / "encoder"
+        encoder_path.mkdir()
+        subdir_path = encoder_path / "subdir"
+        subdir_path.mkdir()
+
+        (folder_path / "config.json").touch()
+        (folder_path / "model.safetensors").touch()
+        (folder_path / "model.bin").touch()  # won't be included
```

Comment:
If these are not included, we're logging that info, right? In that case, should we also test if the info is logged? Sorry if it's already there.

Reply:
There are some `DEBUG` logs yes but not worth testing them IMO

Reply:
Will agree to disagree here but really up to you. No strong onions.